app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,142 @@
+import { delay, Worker } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { setMonthsFromNow } from '~/lib/domains.server';
+import { createRecord, getChangeStatus } from '~/lib/dns.server';
+import { RecordStatus } from '@prisma/client';
+
+import type { DomainRecord } from '~/lib/domains.server';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export const addDbRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    if (!username) {
+      throw new Error('Username is not provided');
+    }
+
+    const existingRecords = await prisma.record.count({
+      where: {
+        type,
+        name,
+        value,
+      },
+    });
+
+    if (existingRecords > 0) {
+      throw new Error('Record already exists');
+    }
+
+    try {
+      const result = await prisma.record.create({
+        data: {
+          username,
+          type,
+          name,
+          value,
+          expiresAt: setMonthsFromNow(6),
+          status: RecordStatus.pending,
+        },
+      });
+      return result.id;
+    } catch (error) {
+      logger.warn('Could not create a record in DB', { error });
```

Comment:
You should rethrow the existing error object here to stop the process and indicate a failure
```suggestion
      logger.warn('Could not create a record in DB', { error });
      throw error;
```

Reply:
I think with Winston (different from pino), you are supposed to use the `error` bare:

```ts
logger.warn('Could not create a record in DB', error);
```

Reply:
Will be updated